gh-113471: Add custom default Content-Type to http.server (#113475)
python/cpython: Doc/library/http.server.rst

      header (using :meth:`send_header`) in all of its responses to clients.
      For backwards compatibility, the setting defaults to ``'HTTP/1.0'``.

>    .. attribute:: default_content_type
> 
>       Specifies the content-type header value sent when the mime-type cannot
>       be guessed from the file extension of the requested URL.
>       For backwards compatibility, the setting defaults to
>       ``'application/octet-stream'``.
> 

Reviewer: ```suggestion
   .. attribute:: default_content_type

      Specifies the content-type header value sent when the MIME type
      cannot be guessed from the file extension of the requested URL.
      By default, it is set to ``'application/octet-stream'``.

      .. versionadded:: next
```

:attr:`index_pages`.

> By default, the server uses the MIME type ``'application/octet-stream'`` for
> the ``Content-Type`` header when the content type cannot be guessed from the
> URL's extension (if any). The option ``--default-content-type`` overrides
> this default with one of the user's choosing::
> 
>         python -m http.server --default-content-type text/html
> 
> .. versionchanged:: next
>    Added the ``--default-content-type`` option.

Reviewer 2: ```suggestion
.. option:: -default-content-type <content_type>

   Specifies the Content-Type HTTP header. By default, the server uses the
   MIME type ``'application/octet-stream'`` for the ``Content-Type`` header
   when the content type cannot be guessed from the URL's extension (if any)::

        python -m http.server --default-content-type text/html

   .. versionadded:: next
```

Reviewer 2: @jcomeauictx we added usage of `.. option::` directory for CLI flags, we also need to move the description of this flag to the right place, could you move it? (I also forgot the second `-` before `default-content-type`, please add it :)

.. versionchanged:: 3.11
   Added the ``--protocol`` option.

By default, the server uses the MIME type ``'application/octet-stream'`` for

> the ``content-type`` header when the content type cannot be guessed from the

Reviewer: ```suggestion
the ``Content-Type`` header when the content type cannot be guessed from the
```

URL's extension (if any). The option ``--default-content-type`` overrides
this default with one of the user's choosing::

        python -m http.server --default-content-type text/html

> .. versionchanged:: 3.13

Reviewer 2: ```suggestion
.. versionchanged:: next
```
